kornia/feature/laf.py

```diff
@@ -0,0 +1,381 @@
+    Used for local features visualization, see visualize_LAF function
+    Args:
+        LAF: (torch.Tensor).
+        n_pts: number of points to output
+
+    Returns:
+        pts: (torch.Tensor) tensor of boundary points
+
+    Shape:
+        - Input: :math:`(B, N, 2, 3)`
+        - Output:  :math:`(B, N, n_pts, 2)`
+    """
+    n_dims = len(LAF.size())
+    if (n_dims != 4):
+        raise TypeError(
+            "LAF shape should be must be [Nx2x3] or [BxNx2x3]. "
+            "Got {}".format(LAF.size())
+        )
+    B, N, _, _ = LAF.size()
+    pts = torch.cat([torch.sin(torch.linspace(0, 2 * math.pi, n_pts - 1)).unsqueeze(-1),
+                     torch.cos(torch.linspace(0, 2 * math.pi, n_pts - 1)).unsqueeze(-1),
+                     torch.ones(n_pts - 1, 1)], dim=1)
+    # Add origin to draw also the orientation
+    pts = torch.cat([torch.tensor([0, 0, 1.]).view(1, 3), 
+                     pts],
+                    dim=0).unsqueeze(0).expand(B * N, n_pts, 3)
+    pts = pts.to(LAF.device).to(LAF.dtype)
+    aux = torch.tensor([0, 0, 1.]).view(1, 1, 3).expand(B * N, 1, 3)
+    HLAF = torch.cat([LAF.view(-1, 2, 3), aux.to(LAF.device).to(LAF.dtype)], dim=1)
+    pts_h = torch.bmm(HLAF, pts.permute(0, 2, 1)).permute(0, 2, 1)
+    return kornia.convert_points_from_homogeneous(
+           pts_h.view(B, N, n_pts, 3))
+
+
+def get_LAF_pts_to_draw(img: torch.Tensor,  # pragma: no cover
+                  LAF: torch.Tensor,
+                  img_idx: int = 0):
+    """
+    Returns numpy array for drawing LAFs
+    """
+    pts = LAF2pts(LAF[img_idx:img_idx + 1])[0]
+    pts_np = pts.detach().permute(1, 0, 2).cpu().numpy()
+    return (pts_np[:, :, 0], pts_np[:, :, 1])
+
+
+def visualize_LAF(img: torch.Tensor,  # pragma: no cover
+                  LAF: torch.Tensor,
+                  img_idx: int = 0,
+                  color = 'r'):
+    """
+    Draws affine regions (LAF)
+    """
+    if MATPLOTLIB:
+        x, y = get_LAF_pts_to_draw(img, LAF,  img_idx)
+        plt.figure()
+        plt.imshow(kornia.utils.tensor_to_image(img[img_idx]))
+        plt.plot(x, y, color)
+        plt.show()
+    return
+
+
+def denormalize_LAF(LAF: torch.Tensor, images: torch.Tensor) -> torch.Tensor:
+    """
+    De-normalizes LAFs from scale to image scale
+    B,N,H,W = images.size()
+    MIN_SIZE = min(H,W)
+    [a11 a21 x]
+    [a21 a22 y]
+    becomes
+    [a11*MIN_SIZE a21*MIN_SIZE x*W]
+    [a21*MIN_SIZE a22*MIN_SIZE y*H]
+    Args:
+        LAF: (torch.Tensor).
+        images: (torch.Tensor) images, LAFs are detected in
+
+    Returns:
+        LAF: (torch.Tensor).
+
+    Shape:
+        - Input: :math:`(B, N, 2, 3)`
+        - Output:  :math:`(B, N, 2, 3)`
+    """
+    n_dims = len(LAF.size())
+    if (n_dims != 4):
+        raise TypeError(
+            "LAF shape should be must be [Nx2x3] or [BxNx2x3]. "
+            "Got {}".format(LAF.size())
+        )
+    n, ch, h, w = images.size()
```

Comment:
since from the whole batch it's only needed height/width. I'd provide to the function only this two parameters intead of the whole image batch

Reply:
Since the data is never copied, I think, it is easier to provice the image batch itself. It is also error-safe from the cases, when you provide some wrong width or height